pkg/cmd/grafana-cli/commands/conflict_user_command.go

```diff
@@ -333,64 +346,54 @@ func (r *ConflictResolver) MergeConflictingUsers(ctx context.Context) error {
 
 		// creating a session for each block of users
 		// we want to rollback incase something happens during update / delete
-		if err := r.Store.WithDbSession(ctx, func(sess *db.Session) error {
-			err := sess.Begin()
-			if err != nil {
-				return fmt.Errorf("could not open a db session: %w", err)
-			}
-			for _, u := range users {
-				if u.Direction == "+" {
-					id, err := strconv.ParseInt(u.ID, 10, 64)
-					if err != nil {
-						return fmt.Errorf("could not convert id in +")
-					}
-					intoUserId = id
-				} else if u.Direction == "-" {
-					id, err := strconv.ParseInt(u.ID, 10, 64)
-					if err != nil {
-						return fmt.Errorf("could not convert id in -")
-					}
-					fromUserIds = append(fromUserIds, id)
-				}
-			}
-			if _, err := sess.ID(intoUserId).Where(sqlstore.NotServiceAccountFilter(r.Store)).Get(&intoUser); err != nil {
-				return fmt.Errorf("could not find intoUser: %w", err)
-			}
-
-			for _, fromUserId := range fromUserIds {
-				var fromUser user.User
-				exists, err := sess.ID(fromUserId).Where(sqlstore.NotServiceAccountFilter(r.Store)).Get(&fromUser)
+		for _, u := range users {
```

Comment:
We need to enclose this block inside a `r.Store.InTransaction(ctx, func(ctx context.Context) error {}` (I guess that was the purpose of the `sess.Begin()`)
In that way a new session [is created](https://github.com/grafana/grafana/blob/ddc3706f19be97c7d9fca80a80a81781b4fe9235/pkg/services/sqlstore/transactions.go#L37) and then it is [stored in the context](https://github.com/grafana/grafana/blob/ddc3706f19be97c7d9fca80a80a81781b4fe9235/pkg/services/sqlstore/transactions.go#L31) so that subsequent `sess.WithDBSession()` calls will [re-use](https://github.com/grafana/grafana/blob/ddc3706f19be97c7d9fca80a80a81781b4fe9235/pkg/services/sqlstore/session.go#L108) the same session.

If tested it (with the `InTransaction()` addition) by adding a new user with a conflicting login (case-insensitively) with another user and run all the following commands successfully:
```
./bin/grafana-cli admin user-manager conflicts list
./bin/grafana-cli admin user-manager conflicts generate-file
./bin/grafana-cli admin user-manager conflicts validate-file /var/folders/qs/_b_z_mm141z19jkf0w2xwtqh0000gp/T/conflicting_user_832113838.diff
./bin/grafana-cli admin user-manager conflicts ingest-file /var/folders/qs/_b_z_mm141z19jkf0w2xwtqh0000gp/T/conflicting_user_832113838.diff```

Reply:
done and tested as well 👍